mitosheet/mitosheet/api/graph/plotly_express_graphs.py

```diff
@@ -0,0 +1,213 @@
+        x_arg = x_axis_column_headers[0]
+    if num_x_axis_column_headers > 1:
+        x_arg = x_axis_column_headers
+
+    y_arg = None
+    if num_y_axis_column_headers == 1:
+        y_arg = y_axis_column_headers[0]
+    if num_y_axis_column_headers > 1:
+        y_arg = y_axis_column_headers
+    
+    if graph_type == BOX:
+        return px.box(df, x=x_arg, y=y_arg)
+    if graph_type == HISTOGRAM:
+        return px.histogram(df, x=x_arg, y=y_arg)
+    if graph_type == BAR:
+        return px.bar(df, x=x_arg, y=y_arg)
+    if graph_type == SCATTER:
+        return px.scatter(df, x=x_arg, y=y_arg)
+
+    
+def graph_creation_code(
+    graph_type: str, 
+    df_name: str, 
+    x_axis_column_headers: List[ColumnHeader],
+    y_axis_column_headers: List[ColumnHeader]
+) -> str:
+    """
+    Returns the code for creating the Plotly express graph
+    """
+    num_x_axis_column_headers = len(x_axis_column_headers)
+    num_y_axis_column_headers = len(y_axis_column_headers)
+
+    # Create the chords that we use to construct the graph
+    x_chord = ''
+    if num_x_axis_column_headers == 1:
+        # Note: In the new interface, x will always have a length of 0 or 1
+        x_chord = f'x="{x_axis_column_headers[0]}"'
+    if num_x_axis_column_headers > 1:
+        x_chord = f'x={x_axis_column_headers}'
+
+    y_chord = ''
+    if num_y_axis_column_headers == 1:
+        y_chord = f'y="{y_axis_column_headers[0]}"'
+    if num_y_axis_column_headers > 1:
+        y_chord = f'y={y_axis_column_headers}'
+
+    # Create the string of all the parameters used to create the graph
+    all_chords = [df_name, x_chord, y_chord]
+    params = (', ').join(list(filter(lambda chord: chord != '', all_chords)))
+
+    if graph_type == BOX:
+        return f"fig = px.box({params})"
+    if graph_type == HISTOGRAM:
+        return f"fig = px.histogram({params})"
+    if graph_type == BAR:
+        return f"fig = px.bar({params})"
+    if graph_type == SCATTER:
+        return f"fig = px.scatter({params})"
+
+
+def graph_styling(fig, graph_type: str, column_headers: List[ColumnHeader], filtered: bool):
+    """
+    Styles the Plotly express graph figure
+    """
+
+    # Create the parameters that we use to style the graph
+    graph_title = get_graph_title(column_headers, [], filtered, graph_type)
+    barmode = get_barmode(graph_type)
+
+    # Actually update the style of the graph
+    fig.update_layout(
+        title = graph_title,
+        barmode=barmode
+    )
+    return fig 
+
+
+def graph_styling_code(graph_type: str, column_headers: List[ColumnHeader], filtered: bool) -> str:
+    """
+    Returns the code for styling the Plotly express graph
+    """
+    graph_title = get_graph_title(column_headers, [], filtered, graph_type)
+    barmode = get_barmode(graph_type)
+
+    # Create the chords that we use to style the graph
+    graph_title_chord = f'title="{graph_title}"' 
```

Comment:
Same comment as above. This is an example where the above approach scales much better.